pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -620,30 +767,43 @@ func (plugin *kubenetNetworkPlugin) syncEbtablesDedupRules(macAddr net.HardwareA
 		return
 	}
 
-	klog.V(3).Infof("Filtering packets with ebtables on mac address: %v, gateway: %v, pod CIDR: %v", macAddr.String(), plugin.gateway.String(), plugin.podCidr)
+	// ensure custom chain exists
 	_, err = plugin.ebtables.EnsureChain(utilebtables.TableFilter, dedupChain)
 	if err != nil {
 		klog.Errorf("Failed to ensure %v chain %v", utilebtables.TableFilter, dedupChain)
 		return
 	}
 
+	// jump to custom chain to the chain from core tables
 	_, err = plugin.ebtables.EnsureRule(utilebtables.Append, utilebtables.TableFilter, utilebtables.ChainOutput, "-j", string(dedupChain))
 	if err != nil {
 		klog.Errorf("Failed to ensure %v chain %v jump to %v chain: %v", utilebtables.TableFilter, utilebtables.ChainOutput, dedupChain, err)
 		return
 	}
 
-	commonArgs := []string{"-p", "IPv4", "-s", macAddr.String(), "-o", "veth+"}
-	_, err = plugin.ebtables.EnsureRule(utilebtables.Prepend, utilebtables.TableFilter, dedupChain, append(commonArgs, "--ip-src", plugin.gateway.String(), "-j", "ACCEPT")...)
-	if err != nil {
-		klog.Errorf("Failed to ensure packets from cbr0 gateway to be accepted")
-		return
+	// per gateway rule
+	for idx, gw := range plugin.podGateways {
+		klog.V(3).Infof("Filtering packets with ebtables on mac address: %v, gateway: %v, pod CIDR: %v", macAddr.String(), gw.String(), plugin.podCIDRs[idx].String())
 
-	}
-	_, err = plugin.ebtables.EnsureRule(utilebtables.Append, utilebtables.TableFilter, dedupChain, append(commonArgs, "--ip-src", plugin.podCidr, "-j", "DROP")...)
-	if err != nil {
-		klog.Errorf("Failed to ensure packets from podCidr but has mac address of cbr0 to get dropped.")
-		return
+		bIsV6 := netutils.IsIPv6(gw)
```

Comment:
since this variable is not used anywhere else other than line 791, maybe the function call should be in-lined instead